deploy/helm/kubecf/templates/_stacks.tpl

```diff
@@ -0,0 +1,137 @@
+  {{- if not $stack.releases }}
+    {{- $_ := set $stack "releases" dict }}
+  {{- end }}
+  {{- if not (index $stack.releases $cc_stack.name) }}
+    {{- $_ := set $stack.releases $cc_stack.name dict }}
+  {{- end }}
+
+  {{- /* *** Create "$stack.install_buildpacks" and "$stack.releases" from the "api" buildpack jobs *** */}}
+  {{- $_ := set $stack "install_buildpacks" list }}
+
+  {{- $_ := include "_config.lookupManifest" (list $ "instance_groups.api.jobs") }}
+  {{- range $job := $.kubecf.retval }}
+    {{- if hasSuffix "-buildpack" $job.release }}
+      {{- $buildpack_shortname := trimSuffix "-buildpack" $job.release }}
+      {{- $_ := set $stack "install_buildpacks" (append $stack.install_buildpacks $buildpack_shortname) }}
+      {{- /* Make sure an entry for the release exists; versions will be filled in by _releases.update */}}
+      {{- if not (index $stack.releases $job.release) }}
+        {{- $_ := set $stack.releases $job.release dict }}
+      {{- end }}
+    {{- end }}
+  {{- end }}
+
+  {{- /* +----------------------------------------------------------------------------------------------+ */}}
+  {{- /* | Setup the additional stacks in .config.stacks and merge their releases into .config.releases | */}}
+  {{- /* +----------------------------------------------------------------------------------------------+ */}}
+
+  {{- $_ := include "_config.lookup" (list $ "stacks") }}
+  {{- range $stack_name, $stack := $.kubecf.retval }}
+    {{- $_ := set $stack "enabled" (has $stack_name $.kubecf.config.install_stacks) }}
+
+    {{- /* *** Update all releases with defaults from stack.releases.$defaults *** */}}
+    {{- $_ := include "_releases.applyDefaults" $stack.releases }}
+
+    {{- /* *** Combine install_buildpacks with prepend and append lists *** */}}
+    {{- $install_buildpacks := list }}
+    {{- if $stack.install_buildpacks_prepend }}
+      {{- $install_buildpacks = $stack.install_buildpacks_prepend }}
+    {{- end }}
+    {{- if $stack.install_buildpacks }}
+      {{- $install_buildpacks = concat $install_buildpacks $stack.install_buildpacks }}
+    {{- end }}
+    {{- if $stack.install_buildpacks_append }}
+      {{- $install_buildpacks = concat $install_buildpacks $stack.install_buildpacks_append }}
+    {{- end }}
+    {{- $_ := set $stack "install_buildpacks" $install_buildpacks }}
+
+    {{- /* create $stack.buildpacks mapping from buildpack shortname to release name */}}
+    {{- $_ := set $stack "buildpacks" dict }}
+
+    {{- $release_prefix := printf "%s-" (default "" $stack.release_prefix) }}
+    {{- $release_suffix := printf "-%s" (default "" $stack.release_suffix) }}
+    {{- range $release_name, $release := $stack.releases }}
+      {{- $_ := set $release "condition" $stack.enabled }}
+
+      {{- /* *** Set buildpack shortname from release name (unless already set) *** */}}
+      {{- if and (hasSuffix "-buildpack" $release_name) (not $release.buildpack) }}
```

Comment:
Yes, I think this is somehow left over from a time when I required the `-buildpack` suffix to differentiate them from rootfs releases, but now that we require that the rootfs release name must match the stack name, this is no longer necessary.

So I would change the code so that `buildpack` is the default value for `install_buildpacks_suffix`. This covers the naming used by CF and SUSE buildpacks, and in the BigCorp example in the docs, the suffix will have to change to `prod-buildpack`.

Does that sounds ok?

Reply:
That seems fine; to reiterate to make sure I'm understanding correctly (since my brain is rather fried from looking at templates):
- The default prefix is the empty string.
- The default suffix is `-buildpack` (after munging).
- If the release name is _not_ the stack name:
  - If the release does not have a `buildpack` key:
    - We set the release's `buildpack` key to (release name without prefix / suffix)

That seems reasonable, since stacks can have a rootfs and some number of buildpacks, and nothing else.

Reply:
Done in 9150f09